solidity/test/isms/ExternalBridgeTest.sol

```diff
@@ -0,0 +1,272 @@
+// SPDX-License-Identifier: MIT or Apache-2.0
+pragma solidity ^0.8.13;
+
+import {Test} from "forge-std/Test.sol";
+import {TypeCasts} from "../../contracts/libs/TypeCasts.sol";
+import {StandardHookMetadata} from "../../contracts/hooks/libs/StandardHookMetadata.sol";
+import {TestMailbox} from "../../contracts/test/TestMailbox.sol";
+import {Message} from "../../contracts/libs/Message.sol";
+import {MessageUtils} from "./IsmTestUtils.sol";
+import {TestRecipient} from "../../contracts/test/TestRecipient.sol";
+import {AbstractMessageIdAuthorizedIsm} from "../../contracts/isms/hook/AbstractMessageIdAuthorizedIsm.sol";
+import {AbstractMessageIdAuthHook} from "../../contracts/hooks/libs/AbstractMessageIdAuthHook.sol";
+
+abstract contract ExternalBridgeTest is Test {
+    using TypeCasts for address;
+    using MessageUtils for bytes;
+
+    uint8 internal constant HYPERLANE_VERSION = 1;
+    uint32 internal ORIGIN_DOMAIN;
+    uint32 internal DESTINATION_DOMAIN;
+    uint256 internal GAS_QUOTE;
+
+    bytes internal unauthorizedHookError;
```

Comment:
I find this a bit weird, can we add this to the return value of `_setExternalOriginSender` or something to make it more explicit instead of requiring children to know to assign to this